src/node_nanomsg.cc

```diff
@@ -117,13 +117,25 @@ NAN_METHOD(Send) {
 
   if (node::Buffer::HasInstance(args[1])) {
     NanReturnValue(NanNew<Number>(nn_send(
-        s, node::Buffer::Data(args[1]), node::Buffer::Length(args[1]), flags)));
+      s, node::Buffer::Data(args[1]), node::Buffer::Length(args[1]), flags)));
   } else {
     v8::String::Utf8Value str(args[1]->ToString());
     NanReturnValue(NanNew<Number>(nn_send(s, *str, str.length(), flags)));
   }
 }
 
+NAN_METHOD(ZerocopySend) {
+  NanScope();
+
+  int s = args[0]->Int32Value();
+  int len = node::Buffer::Length(args[1]);
+
+  void *msg = nn_allocmsg(len, 0);
+  memcpy(msg, node::Buffer::Data(args[1]), len);
```

Comment:
This is great, seeing this helps me think about how we might support zero copy send.

Having a function called ZerocopySend that contains a memcpy is killing me.  I wonder if we could change the interface to only work with a special object we returned that was backed by a call to nn_allocmsg.

I'm not a fan of only supporting buffers and not strings.  If people wanted to go out of their way to have faster sends, then we could make the interface not quite as nice and keep the nicer API (vanilla send) around, too.

For instance, I wonder if we could do something like:

```javascript
var nanomsg = require('nanomsg');
var msg = "hello world";
var buffer = nanomsg.nanomsgBackedBuffer(msg);
nanomsg.zeroCopySend(buffer);
```

hmm maybe not, since that still involves copying the string into the nn_allocmsg provided buffer.  I'll have to think about how we could avoid that memcpy.

My other thought is why don't we just always use this in vanilla send if it's faster?  If the answer is because we skip the conditional send; my response is we want to support strings or buffers for usability.